source/code/plugins/out_oms_blob.rb

```diff
@@ -136,16 +125,42 @@ def get_blob_uri(container_type, data_type, custom_data_type, suffix)
 
       # remove the BOM (Byte Order Marker)
       clean_body = body.encode(Encoding::UTF_8, :invalid => :replace, :undef => :replace, :replace => "")
-      blob_uri_json = JSON.parse(clean_body)
-      return URI.parse( blob_uri_json["Uri"] )
-    end # get_blob_uri
+      return JSON.parse(clean_body)
+    end # request_blob_json
+
+    # get the blob SAS URI and committed blocks from ODS
+    # parameters
+    #   container_type: string. ContainerType of the data
+    #   data_type: string. DataTypeId of the data
+    #   custom_data_type: string. CustomDataType of the CustomLog
+    #   suffix: string. Suffix of the blob
+    # returns:
+    #   URI. blob SAS URI
+    #   string[]. a list of committed blocks
+    def get_blob_uri_and_committed_blocks(container_type, data_type, custom_data_type, suffix)
+      blob_json = request_blob_json(container_type, data_type, custom_data_type, suffix)
+
+      if blob_json.has_key?("Uri")
+        blob_uri = URI.parse(blob_json["Uri"])
+      else
+        @log.error "JSON from BLOB does not contain a URI"
+        blob_uri = nil
+      end
+      if blob_json.has_key?("CommittedBlockList") and !blob_json["CommittedBlockList"].nil?
+        blocks_committed = blob_json["CommittedBlockList"]
+      else
+        blocks_committed = []
+      end
+
+      return blob_uri, blocks_committed
+    end # get_blob_uri_and_committed_blocks
 
     # append data to the blob
     # parameters:
     #   uri: URI. blob URI
     #   msgs: string[]. messages
     #   file_path: string. file path
-    def append_blob(uri, msgs, file_path)
+    def append_blob(uri, msgs, file_path, blocks_committed)
```

Comment:
nevermind, I don't see we read from this file_path, and thought it's the blob_path.  Then I see it's been used as a blob Metadata.